toolkit/tools/imagegen/configuration/partition.go

```diff
@@ -26,6 +28,7 @@ type Partition struct {
 	Start     uint64          `json:"Start"`
 	Flags     []PartitionFlag `json:"Flags"`
 	Artifacts []Artifact      `json:"Artifacts"`
+	Grow      bool
```

Comment:
Is `Grow` needed since the tools already parse `End = 0` as "fill the disk"?

Reply:
Yes, this is needed when the users happen to specify "--grow" in front of "--size=xxx". Without this field, when the parser parses the "--size" field, it will have no knowledge that this partition should fill the disk and instead perform a calculation based on the input size value.

Reply:
Ah, I see. Could we keep that state local to the parser somehow? Feels odd to add an extra field to the schema just to track this since we already have a mechanism to do so (even if its a bit cumbersome).

Reply:
yea I'll add this to the parser module

Reply:
Re-implemented the logic